cmd/dispatch-service/main.go

```diff
@@ -6,20 +6,30 @@ import (
 	"subscription-api/cmd/dispatch-service/internal"
 	"subscription-api/config"
 	store "subscription-api/internal/db"
-	d_store "subscription-api/internal/db/dispatch"
+	"subscription-api/internal/mailing"
 	ds "subscription-api/internal/services/dispatch"
 	g "subscription-api/internal/services/dispatch/grpc"
 	"subscription-api/internal/sql"
 	"subscription-api/pkg/db"
+
+	pb_cs "subscription-api/pkg/grpc/currency_service"
 	pb_ds "subscription-api/pkg/grpc/dispatch_service"
 	"subscription-api/pkg/utils"
 
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
 )
 
 func main() {
 	env := utils.Must(internal.Env())
-	logger := config.InitLogger(env.Mode)
+	logger := config.InitLogger(env.Mode, config.WithProcess("dispatch-service"))
+	defer logger.Flush()
+
+	currencyServiceConn, err := grpc.NewClient(
+		fmt.Sprintf("%s:%s", env.CurrencyServiceAddress, env.CurrencyServicePort),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	utils.PanicOnError(err, "failed to connect to currency service grpc server")
```

Comment:
I have an counterargument to your opinion. 

Due to this [stackoverflow comment](https://stackoverflow.com/a/44505268), `panic` should be used when we want to stop program immediately, but should be ommited in internal packages. `main` function matches this rule, so `panic` can be used here. 

`log.Fatal` is an alternative to `panic`, but it conflicts with principle of graceful shutdown.To prove his violation I wrote some real code samples.

```go
func cleanupFunc() {
	fmt.Println("close connections")
}

func withGracefulShutdown() {
	defer cleanupFunc()
	panic("paniced error")
}

func withoutGracefulShutdown() {
	defer cleanupFunc()
	log.Fatal("fatal error")
}
``` 
This code
```go
func main() {
	defer func() { fmt.Println("close connection") }()
	withoutGracefulShutdown()
}
```
prints next result to console:
```
2024/06/16 15:03:24 fatal error
exit status 1
```

But this code
```go
func main() {
	defer func() { fmt.Println("close connection") }()
	withGracefulShutdown()
}
```
prints next result to console:
```
cleanup
close connection
panic: paniced error

goroutine 1 [running]:
main.withGracefulShutdown()
        /Users/daha/Documents/genesis_kma_school/case_task/main.go:14 +0x48
main.main()
        /Users/daha/Documents/genesis_kma_school/case_task/main.go:24 +0x38
exit status 2
```

Reply:
Nevertheless, thanks to your comment I removed usage of panic everywhere except of files in `cmd` directory.

Reply:
Panic is a boundary case and you should use it in very few cases.

As for me it's good in initialisation of program because panic provides a good stack trace of what's going on.
Using panic with recover in runtime is bad practice cause recover is Very heave operation for your OS.

When your code is running, you should try to catch and handle all the errors you can have, because errors from API/DB is still business errors that usually get near with 5XX server errors.